### PR TITLE
MNT: Run PTH flake test in prep for supporting pathlib (time)

### PR DESCRIPTION
**Pull request by neutrinoceros**
### Description
Extracted from https://github.com/astropy/astropy/pull/16060

@taldcroft already approved this bit in https://github.com/astropy/astropy/pull/16060#pullrequestreview-1888128982

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by neutrinoceros** [comment]
Note that labels `Docs`, `testing` and `utils.iers` can be removed (they were due to commit from the branch this was previously based off)

File: astropy/time/tests/test_update_leap_seconds.py [hunk 1]
```diff
@@ -58,11 +59,13 @@ def test_auto_update_bad_file(self):
             update_leap_seconds(["nonsense"])
 
     def test_auto_update_corrupt_file(self, tmp_path):
+        # FIXME: update_leap_seconds should be able to handle Path objects
         bad_file = str(tmp_path / "no_expiration")
-        with open(iers.IERS_LEAP_SECOND_FILE) as fh:
+        with Path(iers.IERS_LEAP_SECOND_FILE).open() as fh:
             lines = fh.readlines()
```
Comment (pllim) [inline astropy/time/tests/test_update_leap_seconds.py:65]: Can we replace this with `read_text`?

Comment (neutrinoceros) [reply astropy/time/tests/test_update_leap_seconds.py]: done

File: astropy/time/tests/test_update_leap_seconds.py [hunk 2]
```diff
@@ -58,28 +59,27 @@ def test_auto_update_bad_file(self):
...
-            fh.write("\n".join([line for line in lines if not line.startswith("#")]))
+        bad_file = tmp_path / "no_expiration"
+        lines = Path(iers.IERS_LEAP_SECOND_FILE).read_text().splitlines()
+        bad_file.write_text(
+            "\n".join(line for line in lines if not line.startswith("#"))
+        )
 
         with pytest.warns(AstropyWarning, match="ValueError.*did not find expiration"):
-            update_leap_seconds([bad_file])
+            # FIXME: update_leap_seconds should be able to handle Path objects
```
Comment (pllim) [inline astropy/time/tests/test_update_leap_seconds.py:69]: If this is not an issue already, maybe we should open a GitHub issue and link that here?

Comment (neutrinoceros) [reply astropy/time/tests/test_update_leap_seconds.py:69]: yes. I just took a very quick look and it's indeed not trivial to fix, so an issue seems appropriate. I can open it tomorrow (unless you want to beat me to it)

Comment (pllim) [reply astropy/time/tests/test_update_leap_seconds.py:69]: You can probably provide better details. I can wait. Thanks! 😺

Comment (neutrinoceros) [reply astropy/time/tests/test_update_leap_seconds.py:69]: done in #16929 (and linked in a comment here)

Comment (neutrinoceros) [reply astropy/time/tests/test_update_leap_seconds.py:69]: moving this PR to draft so #16931 can be prioritized. This comment can be removed once  #16931 in merged.

Comment (pllim) [reply astropy/time/tests/test_update_leap_seconds.py:69]: The linked PR is merged. FYI.

Comment (neutrinoceros) [reply astropy/time/tests/test_update_leap_seconds.py]: thanks for reminding me ! rebased, and soon-to-be undrafted

**Review by mhvk — APPROVED** [review]
Looks good to me too. Since already indirectly approved by Tom, I'll go ahead and merge.

**Comment by mhvk** [comment]
Actually, I cannot merge since the aarch64 failed -- operation cancelled -- though it does not seem related...

**Comment by pllim** [comment]
> I cannot merge since the aarch64 failed

Merge should not have been disabled because of that job. It is not one of the required jobs for branch protection. 🤔

**Comment by pllim** [comment]
For completeness, the timeout is a separate issue https://github.com/astropy/astropy/issues/16565

**Pull request merged**